Classes/DDFileLogger.m

```diff
@@ -762,7 +764,7 @@ - (NSTimeInterval)rollingFrequency {
 - (void)setRollingFrequency:(NSTimeInterval)newRollingFrequency {
     dispatch_block_t block = ^{
         @autoreleasepool {
-            self->_rollingFrequency = newRollingFrequency;
+            self->_rollingFrequency = MIN(kDDMaxRollingFrequencey, newRollingFrequency);
```

Comment:
```
    NSTimeInterval interval  = LONG_LONG_MAX / NSEC_PER_SEC;
    NSTimeInterval intervalee  = LONG_LONG_MAX + 1; // xcode says about overflow
    NSTimeInterval intervaler = MIN(interval, intervalee); // value is below zero
```

My suggestion:

```
        NSTimeInterval interval  = LONG_LONG_MAX / NSEC_PER_SEC;
        NSTimeInterval intervalee  = LONG_LONG_MAX + 1;
        NSTimeInterval intervaler = MIN(interval, intervalee);
        NSLog(@"hah? %@", @(intervaler));
        NSLog(@"hah abs %@", @(fabs(intervaler)));
    }
    
    {
        NSTimeInterval interval = LONG_LONG_MAX / NSEC_PER_SEC;
        NSTimeInterval intervalee = LONG_LONG_MAX + 1;
        NSTimeInterval intervaler = MIN(interval, fabs(intervalee));
        NSLog(@"nice max %@", @(intervaler));
        NSLog(@"nice max abs %@", @(fabs(intervaler)));
    }
    
    {
        NSTimeInterval interval = LONG_LONG_MAX / NSEC_PER_SEC;
        NSTimeInterval intervalee = LONG_LONG_MIN - 1;
        NSTimeInterval intervaler = MIN(interval, fabs(intervalee));
        NSLog(@"nice min %@", @(intervaler));
        NSLog(@"nice min abs %@", @(fabs(intervaler)));
    }
```

Reply:
But... Could we just use dates instead? Or _IF_ we can dare, we can add optional linking to CoreMedia with CMTime structure inside.